lib/lobby.ts

```diff
@@ -0,0 +1,172 @@
+import { getClient } from '.';
+import { DbItem, IDbItem } from './db-item';
+import { COLLECTION } from './private/enums';
+import { User } from './user';
+
+type DatabaseEntry = Omit<ILobby, 'collectionName'>;
+type ClientResponse = Omit<DatabaseEntry, 'users' | 'uri'>;
+
+export interface LobbyProps {
+  /**
+  * The spotify id of playlist
+  */
+  readonly spotifyPlaylistId?: string;
+
+  /**
+   * The manager of a lobby
+   */
+  readonly managerId: string;
+
+  /**
+   * The participants in a lobby
+   */
+  readonly participants: string[];
+
+  /**
+   * The theme of a lobby
+   */
+  readonly theme: string;
+
+  /**
+   * The playlist name
+   */
+  readonly name: string;
+
+  /**
+   * The list of song ids in the playlist
+   */
+  readonly songIds: string[];
+}
+
+export interface ILobby extends LobbyProps, IDbItem {
+}
```

Comment:
I think we need to define `SpotifyPlaylistId` as a required attribute

Reply:
mm do u think should the playlist be created in the constructor then w/ the spotify api?

Reply:
honestly... i think the constructor should **require** a song playlist..

We can create a **static** function called `create` that is an async function that creates a playlist and passes it into the constructor..

In which case, let's make the constructor `protected`!